rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
You just said "histogram_quantile should work without a by clause". But before you said "I'd expect [...] histogram_quantile(.9, rate(histo[5m])) to error out due to clashing labels". That confuses me.

We have three proposals on the table: 

1. The original from the design doc (`histogram_quantile` as a function - obviously, then it has a semantics that cannot be compared to sum/min/max as sum/min/max are aggregators and not functions).
2. Your proposal. (`histogram_quantile` as an aggregator.) Which I think is fine just that I don't understand why leaving out the by clause should cause an error.
3. My suggested modification of your proposal, where `histogram_quantile` as an aggregator, but - as every aggregator - can be called without a by clause (and then aggregating everything).

Now I'm wondering if you meant 3. from the beginning.